clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -926,6 +941,13 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
   if (D && D->hasAttr<CFICanonicalJumpTableAttr>())
     Fn->addFnAttr("cfi-canonical-jump-table");
 
+  if (getLangOpts().SYCLIsHost) {
+    // Add metadata for a function on Host.
+    if (const FunctionDecl *FD = dyn_cast_or_null<FunctionDecl>(D)) {
+      EmitSubGroupMetadata(FD, Fn);
+    }
+  }
```

Comment:
Don't need braces here.   Suggest changing this to:
```
// Add metadata for a function on Host.

if (getLangOpts().SYCLIsHost)

    if (const FunctionDecl *FD = dyn_cast_or_null<FunctionDecl>(D))

      EmitSubGroupMetadata(FD, Fn);
```

Reply:
I meant here. I would prefer adding the check for the attribute here and then calling EmitSubGroupMetadata